### PR TITLE
MEX-77-FIX

### PR DESCRIPTION
**Pull request by c4role**
## Describe your changes
When you update join list, you shouldn't erase and rewrite, but add user to the existant list.

## Checklist tests
Invite user in your world, try to connect with their login ; add new users and try to connect with the old one and the new one.

## Issue ticket number and link
[MEX-77] https://entsupport.gdapublic.fr/browse/MEX-77

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/i18n/en.json [hunk 1]
```diff
@@ -19,6 +19,13 @@
   "minetest.delete": "Delete",
   "minetest.download.software.link.description": "To get started, download the Minetest software via this link:",
   "minetest.follow": "Followed",
+  "minetest.download.software.link.description": "To get started, download the Minetest software via this link :",
```
Comment (sjollois) [inline src/main/resources/i18n/en.json]: il y a des doublons dedans

File: src/main/java/fr/openent/minetest/service/impl/DefaultWorldService.java [hunk 2]
```diff
@@ -329,6 +370,16 @@ private JsonArray createMailList(UserInfos owner, JsonArray whitelistIdAndLogin,
     public Future<JsonObject> update(String worldId, JsonObject body) {
         Promise<JsonObject> promise = Promise.promise();
 
+        if (body.getBoolean(Field.ISEXTERNAL) == null) {
+            JsonArray whitelist = body.getJsonArray("whitelist");
```
Comment (sjollois) [inline src/main/java/fr/openent/minetest/service/impl/DefaultWorldService.java]: Je mettrais cela dans le if whitelist exists

File: src/main/resources/public/ts/directives/properties-world/properties-world.html [hunk 3]
```diff
@@ -90,7 +90,7 @@ <h3>
         </button>
         <button ng-show="vm.world.isExternal" class="custom-button right-magnet" ng-click="vm.updateImportWorld()"
                 data-ng-disabled="vm.worldForm.title.length < 1 || vm.worldForm.address.length < 1
-                || vm.worldForm.port.length < 1">
+                || vm.worldForm.port.length < 1 || !vm.isPortValid()">
```
Comment (sjollois) [inline src/main/resources/public/ts/directives/properties-world/properties-world.html:93]: il manque le ng-pattern regex du port sur cette page html non ?

**Review by sjollois — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/directives/properties-world/properties-world.html [hunk 4]
```diff
@@ -56,7 +56,7 @@ <h3>
                     <input ng-show="!vm.world.isExternal" type="text" class="seven cell"
                            data-ng-model="vm.worldForm.port" readonly/>
                     <input ng-show="vm.world.isExternal" type="text" class="seven cell"
-                           data-ng-model="vm.worldForm.port"/>
+                           data-ng-model="vm.worldForm.port" pattern="\d{1,5}" title="[[vm.lightbox.portAlert]]"/>
```
Comment (sjollois) [inline src/main/resources/public/ts/directives/properties-world/properties-world.html:59]: il faut instancier vm.lightbox.portAlert

**Review by sjollois — CHANGES_REQUESTED** [review]

File: src/main/java/fr/openent/minetest/service/impl/DefaultWorldService.java (hunk 2 above)
Comment (c4role) [reply src/main/java/fr/openent/minetest/service/impl/DefaultWorldService.java]: c'est fait

File: src/main/resources/public/ts/directives/properties-world/properties-world.html (hunk 4 above)
Comment (c4role) [reply src/main/resources/public/ts/directives/properties-world/properties-world.html:59]: C'est fait

**Review by sjollois — APPROVED** [review]

**Pull request merged**